contracts/staking/StakingPoolInfo.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+library StakingPoolInfo {
+    enum StakingPoolType {
+        FIXED,
+        FLOATING
+    }
+
+    struct RewardToken {
+        address rewardToken;
+        uint32 rewardAmountRatio; // denominated in fixed point ETHER
+        uint256 totalTokenRewardsAvailable; // NOTE: ask cooper whether reward amount is fixed
+    }
+
+    struct StakingPoolData {
+        uint256 daoId;
+        uint128 minTotalPoolStake;
+        uint128 maxTotalPoolStake;
+        uint128 minimumContribution;
+        uint32 epochDuration;
+        uint32 epochStartTimestamp;
+        bool rewardsFinalized;
+        bool emergencyMode;
+        address treasury;
+        uint128 totalStakedAmount;
```

Comment:
this should probably live as a single field in the base contract. it doesnt make sense that you can set it as non zero at initialization at the minimum.